server/authz/authz_test.go

```diff
@@ -0,0 +1,500 @@
+
+		{user: test.UserObserver, object: team, action: read, allow: true},
+		{user: test.UserObserver, object: team, action: write, allow: false},
+	})
+}
+
+func TestAuthorizeLabel(t *testing.T) {
+	t.Parallel()
+
+	label := &kolide.Label{}
+	runTestCases(t, []authTestCase{
+		{user: nil, object: label, action: read, allow: false},
+		{user: nil, object: label, action: write, allow: false},
+
+		{user: test.UserNoRoles, object: label, action: read, allow: true},
+		{user: test.UserNoRoles, object: label, action: write, allow: false},
+
+		{user: test.UserAdmin, object: label, action: read, allow: true},
+		{user: test.UserAdmin, object: label, action: write, allow: true},
+
+		{user: test.UserMaintainer, object: label, action: read, allow: true},
+		{user: test.UserMaintainer, object: label, action: write, allow: true},
+
+		{user: test.UserObserver, object: label, action: read, allow: true},
+		{user: test.UserObserver, object: label, action: write, allow: false},
+	})
+}
+
+func TestAuthorizeHost(t *testing.T) {
+	t.Parallel()
+
+	teamMaintainer := &kolide.User{
+		Teams: []kolide.UserTeam{
+			{Team: kolide.Team{ID: 1}, Role: kolide.RoleMaintainer},
+		},
+	}
+	teamObserver := &kolide.User{
+		Teams: []kolide.UserTeam{
+			{Team: kolide.Team{ID: 1}, Role: kolide.RoleObserver},
+		},
+	}
+	host := &kolide.Host{}
+	hostTeam1 := &kolide.Host{TeamID: ptr.Uint(1)}
+	hostTeam2 := &kolide.Host{TeamID: ptr.Uint(2)}
+	runTestCases(t, []authTestCase{
+		// No access
+		{user: nil, object: host, action: read, allow: false},
+		{user: nil, object: host, action: write, allow: false},
+		{user: nil, object: hostTeam1, action: read, allow: false},
+		{user: nil, object: hostTeam1, action: write, allow: false},
+		{user: nil, object: hostTeam2, action: read, allow: false},
+		{user: nil, object: hostTeam2, action: write, allow: false},
+		{user: test.UserNoRoles, object: host, action: read, allow: false},
+		{user: test.UserNoRoles, object: host, action: write, allow: false},
+		{user: test.UserNoRoles, object: hostTeam1, action: read, allow: false},
+		{user: test.UserNoRoles, object: hostTeam1, action: write, allow: false},
+		{user: test.UserNoRoles, object: hostTeam2, action: read, allow: false},
+		{user: test.UserNoRoles, object: hostTeam2, action: write, allow: false},
+
+		// Global observer can read all
+		{user: test.UserObserver, object: host, action: read, allow: true},
+		{user: test.UserObserver, object: host, action: write, allow: false},
+		{user: test.UserObserver, object: hostTeam1, action: read, allow: true},
+		{user: test.UserObserver, object: hostTeam1, action: write, allow: false},
+		{user: test.UserObserver, object: hostTeam2, action: read, allow: true},
+		{user: test.UserObserver, object: hostTeam2, action: write, allow: false},
+
+		// Global maintainer can read all
+		{user: test.UserMaintainer, object: host, action: read, allow: true},
+		{user: test.UserMaintainer, object: host, action: write, allow: false},
```

Comment:
I believe Global maintainer should be able to write all hosts. 

This is because Global maintainers have the ability to transfer hosts between teams.